Applications/Utils/SimpleMeshCreation/createMeshElemPropertiesFromASCRaster.cpp

```diff
@@ -35,212 +36,158 @@
+    TCLAP::ValueArg<std::string> unit_arg("u",
+                                          "input-unit",
+                                          "input unit of the data",
+                                          true,
+                                          "m/s",
+                                          &allowed_units_constraints);
+    cmd.add(unit_arg);
 
-    cmd.parse( argc, argv );
+    cmd.parse(argc, argv);
 
     // read mesh
-    std::unique_ptr<MeshLib::Mesh> dest_mesh(MeshLib::IO::readMeshFromFile(mesh_arg.getValue()));
+    std::unique_ptr<MeshLib::Mesh> dest_mesh(
+        MeshLib::IO::readMeshFromFile(mesh_arg.getValue()));
 
     // read raster and if required manipulate it
     auto raster = std::unique_ptr<GeoLib::Raster>(
-        FileIO::AsciiRasterInterface::getRasterFromASCFile(raster_arg.getValue()));
-    GeoLib::RasterHeader header (raster->getHeader());
-    if (refinement_arg.getValue() > 1) {
+        FileIO::AsciiRasterInterface::getRasterFromASCFile(
+            raster_arg.getValue()));
+    GeoLib::RasterHeader header(raster->getHeader());
+    if (refinement_arg.getValue() > 1)
+    {
         raster->refineRaster(refinement_arg.getValue());
-        if (refinement_raster_output_arg.getValue()) {
+        if (refinement_raster_output_arg.getValue())
+        {
             // write new asc file
-            std::string new_raster_fname (BaseLib::dropFileExtension(
-                                                  raster_arg.getValue()));
+            std::string new_raster_fname(
+                BaseLib::dropFileExtension(raster_arg.getValue()));
             new_raster_fname += "-" + std::to_string(header.n_rows) + "x" +
                                 std::to_string(header.n_cols) + ".asc";
-            FileIO::AsciiRasterInterface::writeRasterAsASC(*raster, new_raster_fname);
+            FileIO::AsciiRasterInterface::writeRasterAsASC(*raster,
+                                                           new_raster_fname);
         }
     }
 
     // put raster data in a std::vector
     GeoLib::Raster::const_iterator raster_it(raster->begin());
     std::size_t size(header.n_cols * header.n_rows);
     std::vector<double> src_properties(size);
-    for (unsigned row(0); row<header.n_rows; row++) {
-        for (unsigned col(0); col<header.n_cols; col++) {
+    for (unsigned row(0); row < header.n_rows; row++)
+    {
+        for (unsigned col(0); col < header.n_cols; col++)
+        {
             src_properties[row * header.n_cols + col] = *raster_it;
             ++raster_it;
         }
     }
 
-    {
-        double mu, var;
-        std::tie(mu, var) = computeMeanAndVariance(src_properties.begin(), src_properties.end());
-        INFO("Mean value of source: %f.", mu);
-        INFO("Variance of source: %f.", var);
-    }
+    std::unique_ptr<MeshLib::Mesh> src_mesh(
+        MeshLib::RasterToMesh::convert(*raster,
+                                       MeshLib::MeshElemType::TRIANGLE,
```

Comment:
Since `isPointInElementXY` handled only triangle elements. I changed `isPointInElementXY` and now quad elements are handled, too. I'll convert pixels to quads.